spec/registry.md

```diff
@@ -384,6 +384,158 @@ its _resolved value_ contains the implementation-defined integer value
 of the _operand_ of the annotated _expression_,
 together with the resolved options' values.
 
+
+### `:unit` function
+
+The function `:unit` is a selector and formatter for unitized values,
+that is, numeric values associated with a unit of measurement.
+This is a specialized form of numeric selection and formatting.
+
+> [!IMPORTANT]
+> Implementation of this function is **_OPTIONAL_**.
+> Any implementation of this function is strongly encouraged to follow this specification.
+
+#### Operands
+
+The _operand_ of the `:unit` function can be one of any number of
+implementation-defined types,
+each of which contains a numerical `value` plus a `unit` and optionally a `perUnit`.
```

Comment:
I don't understand this sentence. A perunit is a unit. 'unit' includes compounds, so:

kilowatt-hour-per-tonne is a unit, as is

- kilowatt-per-tonne
- per-tonne
- per-kilowatt-hour
- meter-watt-kilogram-per-square-second.

Reply:
I'm mirroring what `MeasureFormat` currently does, which allows for a `perUnit` (so that one can modify a simple unit with a per-unit, like turning "furlongs" into "furlongs per fortnight". I can take this out and we can rely on the per-unit units already defined. It would make things _much_ simpler.

Reply:
For reference: https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/text/MeasureFormat.html#formatMeasurePerUnit-com.ibm.icu.util.Measure-com.ibm.icu.util.MeasureUnit-java.lang.StringBuilder-java.text.FieldPosition-

Reply:
I checked and that is a convenience function. So instead of saying unit='per-tonne', you can say perUnit='tonne'. So if you also want to include it as a 'convenience option' that's fine. I would just define it using that equivalence.

Reply:
No, we're at the minimum requirements level here. I'll let @mihnita define that for ICU (if he wants to expose it) and will remove all the per-unit foolishness.

Reply:
sounds good!